### DISCUSSION TITLE
Leaving boats

### DISCUSSION THREAD

**Discussion by Rqver** [Minestom/Minestom#659] [Help] [answered]
Hi,

I have been able to spawn boats and get them working perfectly, except for the fact I can't leave them - Pressing my crouch key does nothing.

I was told in the Discord that I have to implement the ability to leave boats - does anyone know how?

Thanks

**Comment by MrlnHi** [comment, marked as answer]
Something like this should work:
```
MinecraftServer.getGlobalEventHandler().addListener(PlayerStartSneakingEvent.class, e -> {
    Player player = e.getPlayer();
    Entity vehicle = player.getVehicle();
    if (vehicle != null) {
        vehicle.removePassenger(player);
    }
});
```

**Reply by Rqver** [reply]
Ty :)
